src/network/message.rs

```diff
@@ -223,61 +227,77 @@ impl<N: Network, E: Environment> Message<N, E> {
 
     /// Deserializes the given buffer into a message.
     #[inline]
-    pub fn deserialize(buffer: &[u8]) -> Result<Self> {
-        // Ensure the buffer contains at least the length of an ID.
-        if buffer.len() < 2 {
-            return Err(anyhow!("Invalid message buffer"));
-        }
+    pub fn deserialize<R: Read>(reader: &mut R) -> Result<Self> {
```

Comment:
Should we be clarifying that this `Read` must be "cursored", perhaps requiring the `BufRead` trait?

Reply:
There is a risk that future implementers are unaware and use methods/logic that reconsumes existing bytes. This particular case has cropped up snarkVM serialization before.

Reply:
`BufRead` is an optimisation (e.g. faster reading line by line) that isn't applicable in this specific case. That being said I've extended the trait bound to include `Seek`, which provides the cursoring.